extensions/INTEL/SPV_INTEL_arithmetic_fence.asciidoc

```diff
@@ -0,0 +1,164 @@
+:llvm_intrinsic_link: https://llvm.org/docs/LangRef.html#llvm-arithmetic-fence-intrinsic
+
+SPV_INTEL_arithmetic_fence
+==========================
+
+Name Strings
+------------
+
+SPV_INTEL_arithmetic_fence
+
+Contact
+-------
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+Contributors
+------------
+
+- Dmitry Sidorov, Intel
+- Joe Garvey, Intel
+- Arvind Sudarsanam, Intel
+- Pawel Jurek, Intel
+- Ben Ashbaugh, Intel
+- Greg Lueck, Intel
+
+Notice
+------
+
+Copyright (c) 2024 Intel Corporation.  All rights reserved.
+
+Status
+------
+
+* Complete
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2024-07-16
+| Revision           | 2
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.6 Revision 3.
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+The extension adds *OpArithmeticFenceINTEL* instruction that prevents fast-math
+optimizations between its argument and the expression that contains it.
+For example for fast FP model a compiler can perform reassociation:
+
+If all these *OpFAdd* instructions have the "Fast" FP fast math mode then:
+
+[source,C]
+OpArithmeticFenceINTEL(a + b + c) + d
+
+can be transformed into:
+
+[source,C]
+OpArithmeticFenceINTEL(a + (b + c)) + d
+
+but not into:
+
+[source,C]
+OpArithmeticFenceINTEL(a + b) + (c + d)
```

Comment:
These examples are useful but would IMO be more readable if fully expressed as pseudo SPIR-V assembly (omitting result types):
```
%ab = OpFAdd %a %b
%abc = OpFAdd %ab %c
%abc_fence = OpArithmeticFenceINTEL %abc
%result = OpFAdd %abc_fence %d
```

can be transformed into:

```
%bc = OpFAdd %b %c
%abc = OpFAdd %a %bc
%abc_fence = OpArithmeticFenceINTEL %abc
%result = OpFAdd %abc_fence %d
```

but not into:

```
%ab = OpFAdd %a %b
%ab_fence = OpArithmeticFenceINTEL %ab
%cd = OpFAdd %c %d
%result = OpFAdd %ab_fence %cd
```

Reply:
Updated, thanks!